docs/canary-config.md

```diff
@@ -0,0 +1,99 @@
+Wrapper object that includes the Canary Analysis Configuration and describes how to judge a given metric.
+#### Properties
+- `canary` ([CanaryAnalysisConfiguration](#canary-analysis-configuration))
+
+<a name="canary-analysis-configuration"></a>
+### CanaryAnalysisConfiguration (object)
+Describes how to judge a metric, see the [Netflix Automated Canary Analysis Judge] for more information.
+#### Properties
+- `direction` (enum[string], required) Which direction of statistical change triggers the metric to fail.
+    - `increase` - Use when you want the canary to fail only if it is significantly higher than the baseline (error counts, memory usage, etc, where a decrease is not a failure).
+    - `decrease` - Use when you want the canary to fail only if it is significantly lower than the baseline (success counts, etc, where a larger number is not a failure).
+    - `either` - Use when you want the canary to fail if it is significantly higher or lower than the baseline.
+- `nanStrategy` (enum[string], required) How to handle NaN values which can occur if the metric does not return data for a particular time interval.
+    - `remove` - Use when you expect a metric to always have data and you want the NaNs removed from your data set (usage metrics).
+    - `replace` - Use when you expect a metric to return no data in certain use cases and you want the NaNs replaced with zeros (for example: count metrics, if no errors happened, then metric will return no data for that time interval).
+- `critical` **true** (boolean, optional) - Use to fail the entire canary if this metric fails (recommended for important metrics that signal service outages or severe problems).
+- `mustHaveData` **true** (boolean, optional) - Use to fail a metric if data is missing.
+- `effectSize` ([EffectSize](#effect-size), optional) - Controls how much different the metric needs to be to fail or fail critically.
+
+<a name="effect-size"></a>
+### EffectSize
+Controls the degree of statistical significance the metric needs to fail or fail critically. 
+Metrics marked as critical can also define `criticalIncrease` and `criticalDecrease`. 
+See the [Netflix Automated Canary Analysis Judge] and [Mann Whitney Classifier] classes for more information.
+
+#### Properties
+- `allowedIncrease` **1.1** (number, optional) - Defaults to 1. The multiplier increase that must be met for the metric to fail. This example makes the metric fail when the metric has increased 10% from the baseline.
+- `allowedDecrease` **0.90** (number, optional) - Defaults to 1. The multiplier decrease that must be met for the metric to fail. This example makes the metric fail when the metric has decreased 10% from the baseline.
+- `criticalIncrease` **5.0** (number, optional) - Defaults to 1. The multiplier increase that must be met for the metric to be a critical failure and fail the entire analysis with a score of 0. This example make the canary fail critically if there is a 5x increase.
```

Comment:
@csanden could you verify the accuracy of `criticalIncrease ` and `criticalDecrease` and there relationship with `critical` metrics.

I came up with this from reading the code, and i'm not a Scala expert.

Reply:
IIRC when a metric goes over a critical threshold, the entire test is terminated to minimize negative production impact

Reply:
That is correct. If a metric is marked a `critical` then we will check if the user has supplied a `criticalIncrease` or `criticalDecrease` value. These properties allow the user to control the sensitivity of a "critical" failure. Note that, a critical failure will result in the entire canary to be marked as a failure and receive a score of zero.